src/qibo/base/gates.py

```diff
@@ -1863,3 +1868,158 @@ def __init__(self, q, px=0, py=0, pz=0, seed=None):
+
+    while if :math:`T_1 < T_2`:
+
+    .. math::
+        \\mathcal{E}(\\rho ) = \\mathrm{Tr} _\\mathcal{X}\\left [\\Lambda _{\\mathcal{X}\\mathcal{Y}}(\\rho _\\mathcal{X} ^T \\otimes \\mathbb{I}_\\mathcal{Y})\\right ]
+
+    with
+
+    .. math::
+        \\Lambda = \\begin{pmatrix}
+        1 - p_1 & 0 & 0 & e^{-t / T_2} \\\\
+        0 & p_1 & 0 & 0 \\\\
+        0 & 0 & p_0 & 0 \\\\
+        e^{-t / T_2} & 0 & 0 & 1 - p_0
+        \\end{pmatrix}
+
+    where :math:`p_0 = (1 - e^{-t / T_1})(1 - \\eta )` :math:`p_1 = (1 - e^{-t / T_1})\\eta`
+    and :math:`p_z = 1 - e^{-t / T_1} + e^{-t / T_2} - e^{t / T_1 - t / T_2}`.
+    Here :math:`\\eta` is the ``excited_population``
+    and :math:`t` is the ``time``, both controlled by the user.
+    This gate is based on
+    `Qiskit's thermal relaxation error channel <https://qiskit.org/documentation/stubs/qiskit.providers.aer.noise.thermal_relaxation_error.html#qiskit.providers.aer.noise.thermal_relaxation_error>`_.
+
+    Args:
+        q (int): Qubit id that the noise channel acts on.
+        t1 (float): T1 relaxation time.
+        t2 (float): T2 dephasing time.
+        time (float): the gate time for relaxation error.
+        excited_population (float): the population of the excited state at
+            equilibrium. Default is 0.
+        seed (int): Optional seed for the random number generator when sampling
+            instead of density matrices is used to simulate this gate.
+    """
+
+    def __init__(self, q, t1, t2, time, excited_population=0, seed=None):
+        self.name = "ThermalRelaxationChannel"
+        self.init_args = [q, t1, t2, time]
+        self.init_kwargs = {"excited_population": excited_population,
+                            "seed": seed}
+
+    @staticmethod
+    def _calculate_probs(t1, t2, time, excited_population):
+        import numpy as np
+        if excited_population < 0 or excited_population > 1:
+            raise_error(ValueError, "Invalid excited state population {}."
+                                    "".format(excited_population))
+        if time < 0:
+            raise_error(ValueError, "Invalid gate_time ({} < 0)".format(time))
+        if t1 <= 0:
+            raise_error(ValueError, "Invalid T_1 relaxation time parameter: "
+                                    "T_1 <= 0.")
+        if t2 <= 0:
+            raise_error(ValueError, "Invalid T_2 relaxation time parameter: "
+                                    "T_2 <= 0.")
+        if t2 > 2 * t1:
+            raise_error(ValueError, "Invalid T_2 relaxation time parameter: "
+                                    "T_2 greater than 2 * T_1.")
+
+        p_reset = 1 - np.exp(-time / t1)
+        p0 = p_reset * (1 - excited_population)
+        p1 = p_reset * excited_population
+        if t1 < t2:
+            exp = np.exp(-time / t2)
+        else:
+            rate1, rate2 = 1 / t1, 1 / t2
+            exp = (1 - p_reset) * (1 - np.exp(-time * (rate2 - rate1))) / 2
```

Comment:
I'm trying to find this eq. is this supposed to be `(1 - pz - p0 - p1)`?

Reply:
In this case (T1 > T2) `exp` refers to the first probability in the `probs` list so it is ``pz``. If you expand the product `(1 - p_reset) * (1 - np.exp(...))` you will find that it is equivalent to the equation for pz given in the docs. This is the "original" equation from Qiskit for pz, I just wrote the expanded form in the docs because it was looking better with the symbols I used there.